src/library/treeitem.cpp

```diff
@@ -82,34 +82,22 @@ void TreeItem::initFeatureRecursively(LibraryFeature* pFeature) {
     }
 }
 
-TreeItem* TreeItem::appendChild(
-        std::unique_ptr<TreeItem> pChild) {
-    insertChild(m_children.size(), pChild.get()); // transfer ownership
-    return pChild.release();
-}
-
 TreeItem* TreeItem::appendChild(
         QString label,
         QVariant data) {
-    auto pNewChild = std::make_unique<TreeItem>(
+    std::unique_ptr<TreeItem> pNewChild = std::make_unique<TreeItem>(
             std::move(label),
             std::move(data));
-    return appendChild(std::move(pNewChild));
-}
-
-void TreeItem::removeChild(int row) {
-    DEBUG_ASSERT(row >= 0);
-    DEBUG_ASSERT(row < m_children.size());
-    delete m_children.takeAt(row);
+    TreeItem* pRet = pNewChild.get();
+    insertChild(m_children.size(), std::move(pNewChild));
+    return pRet;
 }
 
-void TreeItem::insertChildren(int row, QList<TreeItem*>& children) {
+void TreeItem::insertChildren(int row, std::vector<std::unique_ptr<TreeItem>>&& children) {
```

Comment:
This is a 2.3 PR where no std::span is available. I also think it is the best solution to just pass the object we have instead of dealing with another type.

Reply:
Mhmm, the absence of span is not great. I guess thats why refactorings don't really belong into the stable branch.
I disagree with the statement that dealing with another type is a bad idea, IMO in this case it would be an improvement because it improves the communication of the intend and also decreases coupling. But since `std::span` is not available (and using iterators is somewhat complicated) I'm fine with keeping it as is. A TODO comment would be nice though.

Reply:
std::span it is a pointer and a size. It has been introduced to manage plain arrays in a safer way. Why would it be better to access a std::vector a like this. The vector has that already. If you like to express that it is borrowed, you can use a const refrence. 

std::span is usefull when you are dealing with only some elements of the vector. But it is useless for all other containers where we have not a consecutive memory layout.

In this case I like that the memory for the unique_ptr itself is also passed to the consuming function. This way you can easy see that the vector with its elements is destroyed.

Reply:
> std::span it is a pointer and a size. It has been introduced to manage plain arrays in a safer way. Why would it be better to access a std::vector a like this.

Well std::span is a generic view over any container that stores its element consecutively in memory. So its generic over a vector and over an array (and possibly QVarLengthArray and many other types that meet that criteria). 

> If you like to express that it is borrowed, you can use a const refrence.

Well not really, I want to express that I want to mutably access the container without growing its size. For that, `const std::span<T>` (note the non-const T) is best. 

> std::span is useful when you are dealing with only some elements of the vector. But it is useless for all other containers where we have not a consecutive memory layout.

Some elements is still just as useful as all elements. Yes if we had a container with a non-consequitive memory layout, then iterators (or ranges) would be best choice, but as I said, that comes with some complexity which you might want to avoid. 

> In this case I like that the memory for the unique_ptr itself is also passed to the consuming function. This way you can easy see that the vector with its elements is destroyed.

Yes thats a fair point, I just don't like the vector because it forces the caller to allocate a vector even though we don't actually care about the fact that its a vector, we just care about the unique_ptr's within it.

Reply:
> Well not really, I want to express that I want to mutably access the container without growing its size. For that, const std::span<T> (note the non-const T) is best.

Ok, that makes sense. I knowledge also that this simplifies the single row case as you pointed out.

Reply:
Thank you. As you pointed out, there is no std::span or gsl::span in 2.3 so can you add a TODO to the function signatures of `TreeItem::insertChildren` and `TreeItemModel::insertTreeItemRows` that reminds us to refactor that in 2.4? Then I'll consider this resolved.